### PR TITLE
Add service generator

### PR DESCRIPTION
**Pull request by guillaumeduboc**

### CODE REVIEW

File: tools/generators/service/index.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable max-lines */
```
Comment (fargito) [inline tools/generators/service/index.ts:1]: Tu peux enlever ça je pense

File: tools/generators/service/typed-json-config/package.json.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    '@swarmion-starter/core-contracts': '1.0.0',
+    '@swarmion-starter/serverless-configuration': '1.0.0',
+    '@swarmion-starter/serverless-helpers': '1.0.0',
+    'aws-sdk': '^2.1084.0',
+    'http-errors': '^2.0.0',
+  },
+  devDependencies: {
+    '@serverless/typescript': '^3.3.0',
+    '@swarmion-starter/configuration': '1.0.0',
+    '@types/http-errors': '^1.8.2',
```
Comment (fargito) [inline tools/generators/service/typed-json-config/package.json.ts:35]: Is this necessary?

**Comment by fargito** [comment]
Do you think there would be a way to replace `@swarmion-starter/` from the tmpl files, for example by using the `npmScope` from nx?

File: tools/generators/service/typed-json-config/package.json.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    'test-linter': 'yarn linter-base-config .',
+    'test-type': 'tsc --noEmit',
+    'test-unit': 'jest --runInBand --collectCoverage --logHeapUsage',
+  },
+  dependencies: {
+    '@swarmion-starter/core-contracts': '1.0.0',
+    '@swarmion-starter/serverless-configuration': '1.0.0',
+    '@swarmion-starter/serverless-helpers': '1.0.0',
+    'aws-sdk': '^2.1084.0',
+    'http-errors': '^2.0.0',
```
[the remark this answers is not in the document]
Comment (guillaumeduboc) [reply tools/generators/service/typed-json-config/package.json.ts]: I removed them as in `backend/core`

File: tools/generators/service/typed-json-config/package.json.ts (hunk 2 above)
Comment (guillaumeduboc) [reply tools/generators/service/typed-json-config/package.json.ts]: I removed them as in `backend/core`

**Comment by guillaumeduboc** [comment]
> Do you think there would be a way to replace `@swarmion-starter/` from the tmpl files, for example by using the `npmScope` from nx?

I replaced all occurence of `@swarmion-starter/` in our templated files 😃

File: tools/generators/service/typed-json-config/tsconfig.json.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  extends: '../../tsconfig.json',
+  compilerOptions: {
+    preserveSymlinks: true,
+    baseUrl: '.',
+    esModuleInterop: true,
+  },
+  references: [
+    { path: '../../packages/configuration' },
+    { path: '../../packages/serverless-configuration' },
+    { path: '../../packages/serverless-helpers' },
```
Comment (adriencaccia) [inline tools/generators/service/typed-json-config/tsconfig.json.ts:13]: As we have seen, we need to point to a different config.
```suggestion
    { path: '../../packages/serverless-helpers/tsconfig.build.json' },
```
Could also make the change in `backend/core/tsconfig.json`, in a different commit ?

**Review by adriencaccia — CHANGES_REQUESTED** [review]

**Review by fargito — APPROVED** [review]
Great job! Thanks a lot :tada:

**Comment by adriencaccia** [comment]
`eslint` is not working

image

**Comment by adriencaccia** [comment]
The `.vscode` symlinked folder is using absolute paths, we need to change it to relative paths
image

**Comment by guillaumeduboc** [comment]
> `eslint` is not working
> 
> image

indeed, eslint is not in the dependencies

**Review by adriencaccia — APPROVED** [review]

**Pull request merged**